src/Rings/mpoly-graded.jl

```diff
@@ -1603,9 +1603,12 @@ function _homogenization(f::MPolyRingElem, S::MPolyDecRing, start_pos::Int)
    return finish(F)
 end
 
-function _homogenization(f::MPolyRingElem, W::ZZMatrix, var::VarName, pos::Int)
+function _homogenization(
+    f::MPolyRingElem, W::ZZMatrix, var::VarName, pos::Int;
+    parent::Union{Nothing, <:MPolyDecRing}=nothing
+  )
     # ASSUME pos is in 1:1+ngens(parent(f))
-  R = parent(f)
+  R = Oscar.parent(f)
   A = copy(symbols(R))
   l = length(A)
   @req pos in 1:l+1 "Index out of range."
```

Comment:
The code below always constructs `A` and `WH`, but they are not used when `parent !== nothing`. We should avoid that.

Note that `var` is not even used in this case. And `pos` is only passed to `_homogenization`.

Perhaps there should then be checked that e.g. `nvars(S) = nvars(R) + 1`... Of course on could then also verify that e.g. weights match, variable names etc. -- or one could deliberately ignore this...


The "clean" alternative would be to have a morphism from `parent(R)` to `S`...